### DISCUSSION TITLE
[SOLVED] How to add trusted proxies?

### DISCUSSION THREAD

**Discussion by MarvvanPal** [nextcloud/all-in-one#3396] [Questions] [answered]
Hi together,

I have set up a Nextcloud AIO using Docker rootless including Nginx Proxy Manager.

Basically everything works. However, I have locked myself out by several incorrect password entries of a user. The bruteforce prevention had struck. I removed the block manually in redis and was able to log in again. The blocked IP address pointed to my Nginx proxy manager.

Unfortunately I was not able to pass the IP to the trusted_proxies via ENV (the log of the mastercontainer mentioned that this is not supported) or configuration.json (no reaction at all). [This manual step](https://github.com/nextcloud/all-in-one#how-to-change-the-domain) was also not successful - "no file or dictionary error" while standing in an empty file...

I found [this post](https://github.com/nextcloud/all-in-one/discussions/2968), but I don't know how to apply this from the mentioned relayd proxy to the Nginx Proxy Manager. Additionally, I can't believe that we can't change the parameter trusted_proxies (and [other parameters](https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/reverse_proxy_configuration.html))?.

How can I add the IP to the array of trusted_proxies?

**Comment by szaimen** [comment, marked as answer]
Hi, usually AIO should take care of this automatically. Can you first follow https://github.com/nextcloud/all-in-one/discussions/2045?

**Comment by MarvvanPal** [comment]
The Nextcloud AIO is running in a IPv4 only network. Even the ISP is providing me only a public IPv4 address. Additionally the Dyndns provider is running IPv4 only. So, I guess we can exclude the IPv6 approach.

**Reply by szaimen** [reply]
Hm... this is interesting. So this reverse proxy container, can you run docker inspect for it?

**Comment by jrstrunk** [comment]
You can edit the config.php, where trusted proxies are set, by running `sudo docker exec -u 0 -it nextcloud-aio-nextcloud /bin/bash`, then `vi config/config.php`. Near the bottom of the file there should be a `trusted_proxies` list.

**Reply by stra1ghtarrow90** [reply]
Hi, I have faced this issue and followed this however the entry I put into the config file reverts back to blank. Is there something here I am missing?

**Reply by jrstrunk** [reply]
Hi, are you making sure to save the changes after making them in vi? You can do this by entering `:wq` after adding the proxy (and not in insert mode). Is the entry being removed after you do something like restart the server, or is it immediately removed before you even exit the bash shell inside the container? You can use `cat config/config.php` to see it's contents without editing.

**Reply by tuetenk0pp** [reply]
I also tried to solve it this way, but the setting gets overwritten after I start up the containers from the AiO Webinterface.

**Reply by pepperoni-pi** [reply]
I edited my config.php and added 172.18.0.1 to the trusted-proxies array and that fixed it for me. Mine is not getting overwritten when I stop/start the containers from the AIO interface.

`sudo nano /var/lib/docker/volumes/nextcloud_aio_nextcloud/_data/config/config.php`

**Reply by J1mooo** [reply]
I have the aio setup in my homelab, not really using it, only ocassionaly when i need to share files out. I noticed every time i tried to login (does not matter what the device is) it gives the error, throttled login, etc. I have a working setup with haproxy running on pfsense -> nextcloud apache and nextcloud always logged the pfsense ip as the source for all traffic.
Since configuring this option it works flawlessly, but did not get an update for the aio container so I'm waiting if it will eventually override it. It would be good if we can set trusted proxies easily like an environment variable

**Reply by rolbx** [reply]
Thanks, i added the IP from the NGINX Proxy Manager an it works!

`  'trusted_proxies' => 
  array (
    0 => '127.0.0.1',
    1 => '::1',
    2 => '192.168.16.5',
`

**Reply by twillfade** [reply]
I had the same problem and added both the internal docker network gateway (it was `172.19.0.1` for me) and the IP of my external reverse proxy and it works!
I haven't tried adding only the reverse proxy yet, so maybe that would be enough.

It seems that the Nextcloud container overwrites the first two values of the `trusted_proxies` array on startup, so be sure to add your custom entries after that. (Maybe that's the problem you're facing @tuetenk0pp)
See the relevant portions of the Nextcloud container startup routine: [https://github.com/nextcloud/all-in-one/blob/b17561d41e575c7efd45fac1a6fb410be79bbe37/Containers/nextcloud/entrypoint.sh#L526-L530](https://github.com/nextcloud/all-in-one/blob/b17561d41e575c7efd45fac1a6fb410be79bbe37/Containers/nextcloud/entrypoint.sh#L526-L530)

**Reply by tuetenk0pp** [reply]
Yes that could be possible. I was able to fix the issue with some advice from @szaimen although I am not able to find the specific comment. 

I just had a look at the `config.php` and I can confirm, adding the reverse proxy as **third** entry does the trick!

**Reply by MarvvanPal** [reply]
Thanks @jrstrunk, this solved my issue, too.

@szaimen as mentioned the solution through editing the json-file didn't worked for me. This manual way editing the config.php worked. I suggest we should mention this in the reverse-proxy.md and/or implement a more convenient way while installing the AIO instance.

**Reply by lucamiani** [reply]
This did not solve the issue for me. I have configured the Trusted Proxies as following: 
`'trusted_proxies' =>
  array (
    0 => '127.0.0.1',
    1 => '::1',
    2 => '192.168.30.25',
    3 => '172.18.0.1',
  ),`

And it doesn't work. When I restart the container I do not see the entry 2 and 3 getting loaded. It only loads the first 2.

`.htaccess has been updated
System config value dbpersistent set to boolean false
System config value files_external_allow_create_new_local set to boolean false
System config value trusted_proxies => 0 set to string 127.0.0.1
System config value trusted_proxies => 1 set to string ::1`

Anyone have an Idea why this is?

**Reply by Zoey2936** [reply]
It doesn't "load" them, the log only shows that value 0 and 1 are set (again) to 127.0.0.1 and ::1

**Comment by A4alli** [comment]
@szaimen . My nextcloud log showing System config value trusted_proxies => 0 set to string 127.0.0.1
 System config value trusted_proxies => 1 set to string ::1
System config value trusted_proxies => 10 set to string 172.18.0.0/16
Config value were not updated

checked conf/conf.php > same values. is it the docker bridge?
